vault/data_source_generic_secret_list.go

```diff
@@ -0,0 +1,52 @@
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/vault/api"
+)
+
+func genericSecretListDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: genericSecretListDataSourceRead,
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Full KV-V1 path where secrets will be listed.",
```

Comment:
Not sure, but this feels a little bit more succinct and generic, we could probably use it in both data sources?
```suggestion
				Description:  "List all secrets under path.",
```